### PR TITLE
Add Java Server Part to Ble Service

### PR DESCRIPTION
**Pull request by ychernysheva**
ToDo: Add Java Server Part to Ble Service

### CODE REVIEW

File: app/src/main/java/org/luxoft/sdl_core/BleCentralService.java [hunk 1]
```diff
@@ -12,22 +12,25 @@
         public static final String TAG = BleCentralService.class.getSimpleName();
         public final static String ACTION_START_BLE = "ACTION_START_BLE";
         public final static String ACTION_STOP_BLE = "ACTION_STOP_BLE";
-        public final static String GET_CLIENT_MESSAGE = "GET_CLIENT_MESSAGE";
         public final static String ON_BLE_READY = "ON_BLE_READY";
-        public final static String ON_MOBILE_REQUEST_RECEIVED = "ON_MOBILE_REQUEST_RECEIVED";
+        public final static String ON_MOBILE_MESSAGE_RECEIVED = "ON_MOBILE_REQUEST_RECEIVED";
```
Comment (AKalinich-Luxoft) [inline app/src/main/java/org/luxoft/sdl_core/BleCentralService.java]:
```suggestion
        public final static String ON_MOBILE_MESSAGE_RECEIVED = "ON_MOBILE_MESSAGE_RECEIVED";
```

File: app/src/main/java/org/luxoft/sdl_core/BleReader.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package org.luxoft.sdl_core;
+
+public abstract class BleReader {
```
Comment (AKalinich-Luxoft) [inline app/src/main/java/org/luxoft/sdl_core/BleReader.java]: @ychernysheva please read about https://www.w3schools.com/java/java_interface.asp
I think in this specific case it is more preferrable than abstract class

File: app/src/main/java/org/luxoft/sdl_core/BleWriter.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package org.luxoft.sdl_core;
+
+public abstract class BleWriter {
```
Comment (AKalinich-Luxoft) [inline app/src/main/java/org/luxoft/sdl_core/BleWriter.java]: @ychernysheva please replace with interface

File: app/src/main/java/org/luxoft/sdl_core/JavaToNativeBleAdapter.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package org.luxoft.sdl_core;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.os.Message;
+
+public class JavaToNativeBleAdapter extends Thread {
+    public static final String TAG = JavaToNativeBleAdapter.class.getSimpleName();
+    final int WRITE_ID = 1;
+    final int READ_ID = 2;
```
Comment (AKalinich-Luxoft) [inline app/src/main/java/org/luxoft/sdl_core/JavaToNativeBleAdapter.java]:
```suggestion
    private static final int WRITE_ID = 1;
    private static final int READ_ID = 2;
```

File: app/src/main/java/org/luxoft/sdl_core/JavaToNativeBleAdapter.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            }
+        };
+        Looper.loop();
+        mWriter.Disconnect();
+        mReader.Disconnect();
+    }
+
+    public void setStopThread(boolean value){
+        //stopThread = value;
+        Thread.currentThread().interrupt();
```
Comment (AKalinich-Luxoft) [inline app/src/main/java/org/luxoft/sdl_core/JavaToNativeBleAdapter.java]:
```suggestion
        mHandler.getLooper().quit();
```

File: app/src/main/java/org/luxoft/sdl_core/BleCentralService.java (hunk 1 above)
Comment (ychernysheva) [reply app/src/main/java/org/luxoft/sdl_core/BleCentralService.java]: Fixed in 8effe4942498195cce7a335540df658a92a3e296

File: app/src/main/java/org/luxoft/sdl_core/BleReader.java (hunk 2 above)
Comment (ychernysheva) [reply app/src/main/java/org/luxoft/sdl_core/BleReader.java]: Done in 8effe4942498195cce7a335540df658a92a3e296

File: app/src/main/java/org/luxoft/sdl_core/BleWriter.java (hunk 3 above)
Comment (ychernysheva) [reply app/src/main/java/org/luxoft/sdl_core/BleWriter.java]: Done in 8effe4942498195cce7a335540df658a92a3e296

File: app/src/main/java/org/luxoft/sdl_core/JavaToNativeBleAdapter.java (hunk 4 above)
Comment (ychernysheva) [reply app/src/main/java/org/luxoft/sdl_core/JavaToNativeBleAdapter.java]: Done in 8effe4942498195cce7a335540df658a92a3e296

File: app/src/main/java/org/luxoft/sdl_core/JavaToNativeBleAdapter.java (hunk 5 above)
Comment (ychernysheva) [reply app/src/main/java/org/luxoft/sdl_core/JavaToNativeBleAdapter.java]: Done in 8effe4942498195cce7a335540df658a92a3e296

**Review by AKalinich-Luxoft — APPROVED** [review]

**Pull request merged**